include/exadg/aero_acoustic/single_field_solvers/analytical_time_int_fluid.h

```diff
@@ -0,0 +1,198 @@
+ */
+template<int dim, typename Number>
+class TimeIntAnalytic : public TimeIntBDF<dim, Number>
+{
+  using Base       = TimeIntBDF<dim, Number>;
+  using VectorType = typename Base::VectorType;
+  using Operator   = SpatialOperatorBase<dim, Number>;
+
+public:
+  TimeIntAnalytic(
+    std::shared_ptr<Operator>                       operator_in,
+    std::shared_ptr<HelpersALE<dim, Number> const>  helpers_ale_in,
+    std::shared_ptr<PostProcessorInterface<Number>> postprocessor_in,
+    Parameters const &                              param_in,
+    std::function<void(VectorType & velocity, VectorType & pressure, double const time)>
+                     interpolate_analytical_solution_in,
+    MPI_Comm const & mpi_comm_in,
+    bool const       is_test_in)
+    : Base(operator_in, helpers_ale_in, postprocessor_in, param_in, mpi_comm_in, is_test_in),
+      interpolate_analytical_solution(interpolate_analytical_solution_in),
+      velocity(this->order),
+      pressure(this->order)
+  {
+  }
+
+  void
+  print_iterations() const final
+  {
+    std::vector<std::string> names = {"Interpolation analytical solution"};
+    std::vector<double>      iterations_avg{0.0};
+    print_list_of_iterations(this->pcout, names, iterations_avg);
+  }
+
+  VectorType const &
+  get_velocity() const final
+  {
+    return velocity[0];
+  }
+
+
+  VectorType const &
+  get_pressure() const final
+  {
+    return pressure[0];
+  }
+
+  VectorType const &
+  get_velocity_np() const final
+  {
+    return velocity_np;
+  }
+  VectorType const &
+  get_pressure_np() const final
+  {
+    return pressure_np;
+  }
+
+private:
+  VectorType const &
+  get_velocity(unsigned int i /* t_{n-i} */) const final
+  {
+    return velocity[i];
+  }
+
+
+  VectorType const &
+  get_pressure(unsigned int i /* t_{n-i} */) const final
+  {
+    return pressure[i];
+  }
+
+  void
+  allocate_vectors() final
+  {
+    Base::allocate_vectors();
+
+    // velocity
+    for(unsigned int i = 0; i < velocity.size(); ++i)
+      this->operator_base->initialize_vector_velocity(velocity[i]);
+    this->operator_base->initialize_vector_velocity(velocity_np);
+
+    // pressure
+    for(unsigned int i = 0; i < pressure.size(); ++i)
+      this->operator_base->initialize_vector_pressure(pressure[i]);
+    this->operator_base->initialize_vector_pressure(pressure_np);
+  }
+
+
+  /**
+   * This function simply sets the analytical
+   */
+  void
+  do_timestep_solve() final
+  {
+    interpolate_analytical_solution(velocity_np, pressure_np, this->get_next_time());
+  }
+
+  void
+  prepare_vectors_for_next_timestep() final
+  {
+    Base::prepare_vectors_for_next_timestep();
+
+    push_back(velocity);
+    velocity[0].swap(velocity_np);
+
+    push_back(pressure);
+    pressure[0].swap(pressure_np);
+  }
+
+  void
+  solve_steady_problem() final
+  {
+    AssertThrow(false, dealii::ExcMessage("Aero-Acustic not implemented for steady problem."));
```

Comment:
This class lives in namespace `ExaDG::IncNS`, so I think the message should be formulated accordingly (e.g. replace "Aero-acoustic" by TimeIntAnalytic).

Reply:
[41de64d](https://github.com/exadg/exadg/pull/659/commits/41de64d7d97686575dd2b736d9e6f32c301fdd2f)